waku/node/jsonrpc/relay/handlers.nim

```diff
@@ -93,18 +94,34 @@ proc installRelayApiHandlers*(node: WakuNode, server: RpcServer, cache: MessageC
         timestamp: msg.timestamp.get(Timestamp(0)),
         ephemeral: msg.ephemeral.get(false)
       )
-    
+
+    # ensure the node is subscribed to the pubsubTopic. otherwise it risks publishing
+    # to a topic with no connected peers
+    if pubsubTopic notin node.wakuRelay.subscribedTopics():
+      raise newException(ValueError, "Failed to publish: Node not subscribed to pubsubTopic: " & pubsubTopic)
+
     when defined(rln):
       if not node.wakuRlnRelay.isNil():
-        let success = node.wakuRlnRelay.appendRLNProof(message, 
+        # append the proof to the message
+        let success = node.wakuRlnRelay.appendRLNProof(message,
                                                       float64(getTime().toUnix()))
         if not success:
-          raise newException(ValueError, "Failed to append RLN proof to message")
-
-    let publishFut = node.publish(topic, message)
+          raise newException(ValueError, "Failed to publish: error appending RLN proof to message")
+
+        # validate the message before sending it
+        let result = node.wakuRlnRelay.validateMessage(message)
```

Comment:
unsure if its ok to use `validateMessage` here, since inside it updates a bunch of metrics and call `updateLog`. Any opinion @rymnc

Reply:
yeah I don't think we should use validateMessage, because if you try publishing the same message it would get stuck since the proofMetadata would exist in the nullifierLog afaik. maybe try using hasDuplicate?

Reply:
Something like this 74d38f530c253c7ab8f9745de7649b42de307d06 ?

Reply:
yes!